instance/tasks.py

```diff
@@ -81,13 +88,20 @@ def spawn_appserver(
 
 
 @db_task()
-def appserver_make_active(appserver_id, active=True):
+def make_appserver_active(appserver_id, active=True, deactivate_others=False):
     """
     Mark an AppServer as active or inactive.
     """
-    logger.info('%s AppServer: ID=%s', "Activating" if active else "Deactivating", appserver_id)
+    # Disallow a request to deactivate this appserver *and* all others.
+    deactivate_others = deactivate_others if active else False
```

Comment:
Note to self: this can be removed in favor of `if active and deactivate_others` below.